packages/noya-state/src/reducers/application.ts

```diff
@@ -113,6 +119,19 @@ export type Action =
   | [type: 'setOpacity', amount: number, mode?: SetNumberMode]
   | [type: 'setFixedRadius', amount: number, mode?: SetNumberMode]
   | [type: `set${StyleElementType}Color`, index: number, value: Sketch.Color]
+  | [
+      type: 'setSwatchColor', 
+      swatchId: string | string[] | undefined,
```

Comment:
`undefined` isn't a valid option here, since we need at least 1 swatch selected to change its color

Reply:
it only takes`string | string[]` now